### PR TITLE
[DSM] PEPPER-853 sm id fix

### PR DESCRIPTION
**Pull request by pegahtah**
## Context

I added a check for zero received sm ids.
I trired using varioable assignment but it doesn't work in java 

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

**Review by denniscunningham — COMMENTED** [review]
We should plan to refactor this query (perhaps another ticket?), even if that means executing more than one query, since the logic of this query is lost in the complexity.

For this case (PEPPER-853), is in possible to add logic to check the query results rather than adding more logic (and complexity) to the query?

**Comment by pegahtah** [comment]
@denniscunningham I simplified the query to just check  if the numbers entered for both USS and scrolls count are simultaneously 0 or NULL and not show those tissues.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercurySampleDao.java [hunk 1]
```diff
@@ -28,6 +28,7 @@ public class MercurySampleDao implements Dao<MercurySampleDto> {
                     + "LEFT JOIN ddp_onc_history_detail oD on (mr.medical_record_id = oD.medical_record_id AND NOT oD.deleted <=> 1) "
                     + "LEFT JOIN ddp_tissue t on (oD.onc_history_detail_id = t.onc_history_detail_id AND NOT t.deleted <=> 1) "
                     + "WHERE oD.tissue_received IS NOT NULL AND p.ddp_participant_id = ? AND ddp.instance_name = ? "
+                    + "AND (IFNULL(t.uss_count, 0) <> 0 OR IFNULL(t.scrolls_count, 0) <> 0) "
```
Comment (pegahtah) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercurySampleDao.java:31]: Just avoiding the ones that are both 0 or both null should do the trick.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercurySampleDao.java [hunk 2]
```diff
@@ -40,7 +41,7 @@ public class MercurySampleDao implements Dao<MercurySampleDto> {
                     + "From ddp_mercury_sequencing where ddp_participant_id = ? group by (tissue_id) ) as table2 "
                     + "on table2.seqt = table1.tissue_id "
                     + "LEFT JOIN (select order_id, mercury_sequencing_id, order_status, order_date, mercury_pdo_id "
-                    + "FROM ddp_mercury_sequencing WHERE ddp_participant_id = ?) as table3 ON(table3.mercury_sequencing_id = table2.ms_id)";
+                    + "FROM ddp_mercury_sequencing WHERE ddp_participant_id = ?) as table3 ON(table3.mercury_sequencing_id = table2.ms_id) ";
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercurySampleDao.java:44]: Is this change intentional?

**Review by denniscunningham — APPROVED** [review]
Looks good Pegah. Thx for the changes!
Just left one comment.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercurySampleDao.java (hunk 2 above)
Comment (pegahtah) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercurySampleDao.java:44]: Yeah I like to have an extra space at the end in case we need to concat the query with some other condition
